http-server/src/main/java/com/facebook/airlift/http/server/SecureServlet.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.http.server;
+
+import com.google.common.collect.ImmutableSet;
+
+import javax.annotation.security.RolesAllowed;
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.Principal;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.airlift.http.server.HttpServerConfig.AuthorizationPolicy;
+import static com.google.common.io.ByteStreams.copy;
+import static com.google.common.io.ByteStreams.nullOutputStream;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class SecureServlet
+        extends HttpServlet
+{
+    private final Servlet delegate;
+    private final Authorizer authorizer;
+    private final AuthorizationPolicy authorizationPolicy;
+    private final Set<String> defaultAllowedRoles;
+    private final Optional<Set<String>> allowedRoles;
+
+    public SecureServlet(Servlet delegate,
+            Authorizer authorizer,
+            AuthorizationPolicy authorizationPolicy,
+            Set<String> defaultAllowedRoles)
+    {
+        this.delegate = requireNonNull(delegate, "delegate is null");
+        this.authorizer = requireNonNull(authorizer, "authorizer is null");
+        this.authorizationPolicy = requireNonNull(authorizationPolicy, "authorizationPolicy is null");
+        this.defaultAllowedRoles = requireNonNull(defaultAllowedRoles, "defaultAllowedRoles is null");
+        this.allowedRoles = getRolesFromClassMetadata(delegate);
+    }
+
+    @Override
+    public void init()
+            throws ServletException
+    {
+        super.init();
+        delegate.init(this.getServletConfig());
+    }
```

Comment:
In order to access `initParameters` from the delegate servlet. I've added this: https://github.com/prestodb/airlift/pull/17/files#diff-47e526a2479bc0cbe59875706a697658R62-R68